### DISCUSSION TITLE
Logout function with Keycloak

### DISCUSSION THREAD

**Discussion by 1itt1eB0y** [outline/outline#3672] [Self Hosting] [answered]
I used keycloak as my OIDC, and how I configurate it to make `Logout` in Outline can truly Logout?
When I click `Login`, it will always auto login with the same account.

**Comment by tommoor** [comment]
This seems like it would be configured on Keycloak's side, there isn't anything special within Outline on this front. Please make sure to post the answer if you find it!

**Reply by deniscuculic** [reply]
OIDC_LOGOUT_URI would be really nice to have.

**Comment by 1itt1eB0y** [comment, marked as answer]
Here is the solution:
Realm Settings -> Tokens -> SSO Session Idle
Set its value as you want.
This function allows you to log in again if you have not communicated with keycloak for a set period of time. You can login with other account by token timeout.
It will be better if there is an API for OIDC logout.

**Comment by beatstream69** [comment]
Need option to manually specify **OIDC_LOGOUT_URI** in app config.
Then for IdP like Keycloak you can normally terminate your session by navigating user to that url after internal app logout.
It is [standard OIDC feature](https://openid.net/specs/openid-connect-session-1_0-17.html)
```
https://<KC_URL>/realms/<REALM>/protocol/openid-connect/logout?client_id=<OUTLINE_CLIENTID>&post_logout_redirect_uri=https%3A%2F%2F<OUTLINE_URL>%2F
```

**Comment by deniscuculic** [comment]
Just a workaround for 5 minute session timeout for OIDC

 if ($cookie_sessionid != ""){
                add_header Set-Cookie "sessionid=$cookie_sessionid;path=/;Max-Age=300";
}

**Comment by ksaadDE** [comment]
https://github.com/outline/outline/issues/6909

**Reply by tommoor** [reply]
As discussed in the attached issue `OIDC_LOGOUT_URI` was introduced as an option for post-logout redirect

**Comment by thammawat-s** [comment]
Info from #6909 above,  this works for me.

`OIDC_LOGOUT_URI=https://{your-keycloak-domain}/realms/{your-realm}/protocol/openid-connect/logout?client_id={account-ID-you-created-under-your-realm}`
e.g.
`OIDC_LOGOUT_URI=https://keycloak.example.com/realms/outline/protocol/openid-connect/logout?client_id=outline`
